File: Nth_To_Last_Node_In_List.cpp
/**
 * Definition of ListNode
 * class ListNode {
 * public:
 *     int val;
 *     ListNode *next;
 *     ListNode(int val) {
 *         this->val = val;
 *         this->next = NULL;
 *     }
 * }
 */
class Solution {
public:
    /**
     * @param head: The first node of linked list.
     * @param n: An integer.
     * @return: Nth to last node of a singly linked list. 
     */
    ListNode *nthToLast(ListNode *head, int n) {
        // write your code here
        ListNode* dummy = new ListNode(-1);
        dummy->next = head;
        ListNode* front = dummy;
        ListNode* back = dummy;
        while(n-- > 0) front = front->next;
        while(front){
            back = back->next;
            front = front->next;
        }
        return back;
    }
};

File: Nth to Last Node in List.py
"""
Definition of ListNode
class ListNode(object):

    def __init__(self, val, next=None):
        self.val = val
        self.next = next
"""
class Solution:
    """
    @param head: The first node of linked list.
    @param n: An integer.
    @return: Nth to last node of a singly linked list. 
    """
    def nthToLast(self, head, n):
        # write your code here
        if not head:
            return head
        dummy = ListNode(0)
        dummy.next = head
        fast = dummy
        for __ in range(n):
            fast = fast.next
        
        slow = dummy
        while fast.next:
            fast, slow = fast.next, slow.next
        return slow.next